src/code_file_manager.rb

```diff
@@ -1,8 +1,9 @@
+require 'tempfile'
+
 module CodeFileManager
-  def next_file_absolute_path
-    possible_letters = [('a'..'z'), ('A'..'Z')].map(&:to_a).flatten
-    random_string = (0..10).map { possible_letters[rand(possible_letters.size)] }.join
-    "#{Dir.pwd}/codes/#{random_string}"
+  def next_file_absolute_path(extension)
+    tempfile = Tempfile.create(['foo', extension], "#{Dir.pwd}/codes")
```

Comment:
I don't know how it would prevent the race condition because we are not going to use the file instance only in the server process. We need also to leave it so that other process "linter" can access it. A good thing about temp file is that each time it gets called, it creates a new file with different name. So each time we call "perfom_lint" another file  with completely different name is created, then there is no race condition between parallel calls to the server. The name is just a random name depending on the time of the computer just as we had before